### ISSUE TITLE
Support extend/increase the VM disk size

### ISSUE THREAD

**Issue by marcoczen** [cockpit-project/cockpit-machines#358]
Hi,

Currently is it possible to expand the VM Size ?
Say i start out at 10GB  but then need to make it 50GB ?

Is there  a way to do this without data loss ?

**Comment by KKoukiou** [comment]
Cockpit Machines does not offer this functionality for the time being. You can perform this from CLI like `qemu-img resize vm-image` or `virsh blockresize your-vm vm-image`

Then you need to login in the VM the resize your partitions. 

I will leave the issue open as an RFE  - or if not interested feel free to close it.

**Comment by marcoczen** [comment]
Hi ...

> Cockpit Machines does not offer this functionality for the time being. You can perform this 
> from CLI like `qemu-img resize vm-image` or `virsh blockresize your-vm vm-image`
> 

At least that's possible. Great !

> Then you need to login in the VM the resize your partitions.
> 

What do u mean ? Say I have an Ubuntu Guest ....what will be the next step after the above step is done ?
Run gparted ?


> I will leave the issue open as an RFE - or if not interested feel free to close it.

Pls do consider adding this feature. 

Thanks.

**Comment by marcoczen** [comment]
Thank u. Closed.

**Comment by Thulium-Drake** [comment]
This is still a useful feature to add to the backlog, and as both of the issues mentioning this have been closed, I wouldn't call this complete as most VM managers provide a means to do this.

As far as automatically expanding the guests' filesystems, I'd consider that out of scope for cockpit. To summarize it as a userstory :-) :

As a User of Cockpit I'd like to have the ability to manage the size of the disk images used by the VMs I have created with Cockpit so I can extend them (online or offline) without having to resort to commandline tools.
